### PR TITLE
Update examples for axisgrid1

### PR DESCRIPTION
**Pull request by divyam3897**
Add description for examples for axisgrid1/demo_axis_*
Extension to https://github.com/matplotlib/matplotlib/issues/8921

### CODE REVIEW

File: examples/axes_grid1/demo_axes_divider.py [hunk 1]
```diff
@@ -3,6 +3,7 @@
 Demo Axes Divider
 =================
 
+Example using Axes divider to calculate location of axes and create a divider for them using exisiting axes instances.
```
Comment (NelleV) [inline examples/axes_grid1/demo_axes_divider.py]: This line isn't pep8: it's too long.

You can probably configure your text editor to show you the lines that don't follow pep8 automatically.

File: examples/axes_grid1/demo_axes_divider.py [hunk 2]
```diff
@@ -3,6 +3,8 @@
 Demo Axes Divider
 =================
 
+Example using Axes divider to calculate location of axes
```
Comment (choldgraf) [inline examples/axes_grid1/demo_axes_divider.py]: I think this needs to be one line for sphinx-gallery to parse it properly. I could be wrong though... @NelleV do you know?

Comment (NelleV) [reply examples/axes_grid1/demo_axes_divider.py]: Nope, it doesn't.

**Comment by jklymak** [comment]
My concerns still stand.  “Example demonstrating....” is (doubly!) redundant. “Demo Axes Divider” should just be “Axes Divider” etc.

**Comment by choldgraf** [comment]
This looks good to me - agree that it'd be better if "demo" or "example" weren't in there, but IMO we can handle those things in a subsequent blanket PR that makes this fix for all examples.

**Comment by divyam3897** [comment]
@choldgraf true, there are many examples having that redundancy, it should be uniform and no example should have them in that case. I can remove it in this PR only or a separate PR later for all of them whatever seems right?

**Comment by jklymak** [comment]
@divyam3897  You can remove the redundancy from the caption text, and we can worry about the redundancy in the titles later is what I think @choldgraf means.   Thanks!

**Comment by NelleV** [comment]
We have a lot of examples, so I wouldn't attempt to do this all at once.

What is important is to have meaningful titles and description of the content of the example. "Example demonstrating Hbox Divider to arrange subplots." doesn't really describe what Hbox divider does or how it helps arrange subplots for example.

I'm fine merging this as is, as it is an improvement on the current state of the gallery (thanks for the patch!), but in general, I'd rather have less examples or changes, but more meaningful one.

**Comment by divyam3897** [comment]
@jklymak @NelleV @choldgraf I made the changes, let me know if I can improve them more 👍

**Comment by jklymak** [comment]
Thanks a lot @divyam3897 

So just for reference, I think these descriptions at the top of examples can be as long as we like?  If so, I don't think it hurts future ones to be a bit more informative if warranted (without them becoming a full-on tutorial)

**Comment by divyam3897** [comment]
@jklymak so a 2-3 lines of description would be better?

**Comment by jklymak** [comment]
@divyam3897  Not sure - I'm relatively new too, but I don't think it would hurt.  If you are going through a few more of these, and the example is complicated, feel free to use a couple of lines.  We can always edit it back.

**Comment by choldgraf** [comment]
Yep, I was mistaken in thinking the first sentence could only be one line.
That said, the first sentence should be short and to the point. Longer
descriptions should come in the paragraphs after the first sentence.

**Comment by QuLogic** [comment]
@tacaswell not backporting automatically?

**Comment by tacaswell** [comment]
@meeseeksdev backport to v2.1.0-doc

**Pull request merged**